File: explori/src/Models/Item.js
/*
Item.js

Creates the Item model. Contains the image and description of an item

*/

//root model
import Model from "./Model";

//util lib
import _ from "lodash";
import uid from "uid";

class Item extends Model {
  //constructs the object
  constructor(
    id = uid(),
    collection,
    title,
    description,
    author,
    image_link,
    type = null,
    completed = null,
    selected_text = null,
    extra_options = null
  ) {
    super({
      id,
      collection,
      title,
      description,
      author,
      image_link,
      type,
      completed,
      selected_text,
      extra_options
    });
  }

  static parse(raw_item) {
    return Model.parse(raw_item, Item);
  }

  static parseList(raw_items) {
    return Model.parseList(raw_items, Item);
  }
}

export default Item;

File: scraper/models/Item.py
#
# ./models/Item.py
# Item Model
#
# A item has a title, description, author, location, image_link
#

import json

class Item:

    # item constructor
    def __init__(self, title, description, author, location, image_link, type=None):
        self.title = title
        self.author = author
        self.location = location
        self.description = description
        self.image_link = image_link
        self.type = type

    # convert model to dictionary
    def __dict__(self):

        return {
            "title": self.title,
            "author": self.author,
            "location": self.location,
            "description": self.description,
            "image_link": self.image_link,
            "type": self.type
        }

    # return json version of object
    def __str__(self):
        dict = self.__dict__()
        return json.dumps(dict)

    # string version of Item model
    def __json__(self):
        return self.__str__()
